include/gsl/util

```diff
@@ -52,40 +52,29 @@ template <class F>
 class final_action
 {
 public:
-    static_assert(!std::is_reference<F>::value && !std::is_const<F>::value &&
-                      !std::is_volatile<F>::value,
-                  "Final_action should store its callable by value");
+    template <class FF>
```

Comment:
what's the purpose of this nested template?

Reply:
I missed that this was a template - my brain autocorrected this to a non-template taking `F&&` as would be consistent with the "`F` is sometimes an lvalue reference type" approach. If the intent is that `F` is sometimes an lvalue reference referring to the lvalue argument to `finally`, then this should become:
```c++
explicit final_action(F&& ff) noexcept : f{std::forward<F>(ff)} { }
```
if the intent is that `F` is always an object type, a template constructor is still not a great idea. Sure, it will work with both `const F&` and `F&&` as we'd like, but it will also compete with the copy and move constructors which we don't want. For that case I'd replace this with two constructors:
```c++
explicit final_action(const F& ff) noexcept : f{ff} { }
explicit final_action(F&& ff) noexcept : f{std::move(ff)} { }
```

Reply:
Thanks, I'll take the overload set and avoid the template.